src/main/java/view/DashboardView.java

```diff
@@ -81,93 +96,201 @@ public DashboardView(DashboardViewModel dashboardViewModel,
-                            viewManagerModel.firePropertyChanged();
-                            System.out.println("History button pressed");
-                            // TODO: Do we then need to call GetTransactionHistoryController.execute()
-                            //  to obtain the needed information?
-                        }
-                    }
+                evt -> {
+                    historyViewModel.firePropertyChanged();
+                    viewManagerModel.setActiveView(historyViewModel.getViewName());
+                    viewManagerModel.firePropertyChanged();
                 }
         );
 
-        // NOTE: may or may not be necessary
-        this.setLayout(new BoxLayout(this, BoxLayout.Y_AXIS));
-
         // Make the various display items visible to the user on the display.
-        this.add(title);
-        this.add(welcome);
-        this.add(buttons);
+        this.setLayout(new BoxLayout(this, BoxLayout.X_AXIS));
+        setLayout(new BorderLayout());
+        GridBagConstraints gbc = new GridBagConstraints();
+
+        topPanel = new JPanel(new GridBagLayout());
+        JPanel intermediatePanel1 = new JPanel(new GridBagLayout());
+        middlePanel = new JPanel(new GridBagLayout());
+        JPanel intermediatePanel2 = new JPanel(new GridBagLayout());
+        bottomPanel = new JPanel(new GridBagLayout());
+
+        gbc.gridx = 0;
+        gbc.gridy = 0;
+        gbc.anchor = GridBagConstraints.WEST;
+//        buttons.setPreferredSize(new Dimension(30, 80));
+        topPanel.add(buttons, gbc);
+
+        gbc.gridx = 1;
+        gbc.weightx = 1;
+        gbc.fill = GridBagConstraints.HORIZONTAL;
+        topPanel.add(Box.createHorizontalGlue(), gbc);
+
+        // Spice up the labels a bit.
+        Font labelFont = balanceLabel.getFont();
+//        balanceLabel.setForeground(new Color(14, 20, 138));
+        balanceLabel.setSize(25, 5);
+        balanceLabel.setFont(labelFont.deriveFont(Font.BOLD));
+
+        // Spice up the actual balance (field) as well.
+        Font fieldFont = balanceField.getFont();
+        balanceField.setForeground(Color.DARK_GRAY);
+        balanceField.setFont(fieldFont.deriveFont(Font.ITALIC));
+
+        gbc.gridx = 2;
+        gbc.gridy = 0;
+        gbc.weightx = 0;
+        gbc.fill = GridBagConstraints.NONE;
+        topPanel.add(balanceLabel, gbc);
+
+        gbc.gridx = 3;
+        gbc.gridy = 0;
+        gbc.anchor = GridBagConstraints.EAST;
+        gbc.weightx = 0;
+        topPanel.add(balanceField, gbc);
+
+        // Get DashboardState after running the use case
+        dashboardController.execute();
+        DashboardState state = dashboardViewModel.getState();
+
+        HashMap<String, String> userStats = state.getUserStats();
+        if (userStats != null) {
+//            userStatsTable.setPreferredSize(new Dimension(100, 200));
+            userStatsTable.setModel(new TableModel(userStats));
```

Comment:
this displays daysSinceLastTopup = -1.0
first, change all the keys to English, so "Days Since Last Topup" and make it so if the days since last topped up is <0, then display "Never Topped Up".

balance -> Balance,
totalAssets -> Total Assets (maybe a short description if you want)

Reply:
Bruh, changing those keys is going to be so much work... but I guess we got to do it.

Changing those names which are buried behind maps is going to be a hassle but I suppose it is indeed worth it.